dhis2/dhis2-core
# feat: enforce email verification in system settings

# Summary
Adds a new boolean system setting: `enforceVerifiedEmail`. This settings will redirect users that has not verified their emails, to their account page after login.
The users account page will then show a warning dialog box and a button saying, you need to click the button to send an email verification mail to your registered email account. You then need to click the verification link in the mail to finalize the verification. When done so, you will not be redirect to the account page after login.

### Automatic test:
AuthTest#testRedirectAccountWhenVerifiedEmailEnforced()

### Manual test:
1. Enable `enforceVerifiedEmail` in system settings, UI update is coming in another FE PR, to manually enable with API: `curl -X POST --location "https://play.im.dhis2.org/dev/api/42/systemSettings/keySelfRegistrationNoRecaptcha" \
    -H "Content-Type: text/plain" \
    -d 'true' \
    --basic --user admin:district`
2. Login with any user, that has not verified email. Default admin user should work.
3. Observe that you end up on the URL: "/dhis-web-user-profile/#/account"
